iothub/device/samples/getting started/RoutingTutorial/SimulatedDevice/Program.cs

```diff
@@ -0,0 +1,192 @@
+//  in this article: https://docs.microsoft.com/azure/iot-hub/tutorial-routing
+//The scripts for creating the resources are included in the resources folder in this
+//  Visual Studio solution. 
+//
+// This program encodes the message body so it can be queried against by the Iot hub.
+//
+// If you want to read an encoded message body, you can do this:
+//   * route the message to storage,
+//   * retrieve the message from the storage account by downloading the blob,
+//   * use the ReadOneRowFromFile method at the bottom of this module to read the first row of the file.
+//   * It will read the file, decode the first row in the file, and write it out to a new file 
+//       in ASCII so you can view it.
+
+using CommandLine;
+using Microsoft.Azure.Devices.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SimulatedDevice
+{
+    internal class Program
+    {
+        private static DeviceClient s_deviceClient;
+
+        private static async Task Main(string[] args)
+        {
+            // Parse application parameters
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams =>
+                {
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
+
+            // If this is false, it will submit messages to the iot hub. 
+            // If this is true, it will read one of the output files and convert it to ASCII.
+            if (parameters.ReadTheFile)
+            {
+                // If you want to decode an output file, put the path in ReadOneRowFromFile(), 
+                //   uncomment the call here and the return command, then run this application.
+                ReadOneRowFromFile();
+            }
+            else
+            {
+                // Send messages to the simulated device. Each message will contain a randomly generated 
+                //   Temperature and Humidity.
+                // The "level" of each message is set randomly to "storage", "critical", or "normal".
+                // The messages are routed to different endpoints depending on the level, temperature, and humidity.
+                //  This is set in the tutorial that goes with this sample: 
+                //  http://docs.microsoft.com/azure/iot-hub/tutorial-routing
+
+                Console.WriteLine("Routing Tutorial: Simulated device\n");
+                string myDeviceId = parameters.DeviceId;
+                string deviceKey = parameters.DeviceKey;
+                s_deviceClient = DeviceClient.Create(
+                    parameters.IotHubUri,
+                    new DeviceAuthenticationWithRegistrySymmetricKey(myDeviceId, deviceKey),
```

Comment:
nit: should object parameters on line 69 be indented once more time to make it easier to read?